src/processors/vestingContractProcessor.ts

```diff
@@ -114,79 +104,77 @@ export const processVestingContract = async (
 
-  const ixs: TransactionInstruction[] = [ComputeBudgetProgram.setComputeUnitLimit({ units: 220_000 })];
+  const ixs: TransactionInstruction[] = [ComputeBudgetProgram.setComputeUnitLimit({ units: SIMULATION_CU })];
   if (computePrice) {
     ixs.push(ComputeBudgetProgram.setComputeUnitPrice({ microLamports: computePrice }));
   }
   ixs.push(ix);
-  const messageV0 = new TransactionMessage({
-    payerKey: sender.publicKey,
-    recentBlockhash: recentBlockInfo.blockhash,
-    instructions: ixs,
-  }).compileToV0Message();
-  const tx = new VersionedTransaction(messageV0);
-  tx.sign([sender, metadata]);
 
-  for (let i = 0; i < 3; i++) {
-    const res = await connection.simulateTransaction(tx, { commitment });
-    if (res.value.err) {
-      const errMessage = res.value.err.toString();
-      if (!errMessage.includes("BlockhashNotFound") || i === 2) {
-        throw new Error(errMessage);
-      }
+  while (true) {
+    const contractId = await fetchExistingStream(connection, pid, mint, recipientInfo.address);
+    if (contractId) {
+      console.log(`Recipient ${recipientInfo.address.toBase58()} already has a stream for this mint, will skip`);
+      return { txId: "", contractId: contractId };
     }
-    break;
-  }
 
-  let signature = bs58.encode(tx.signatures[0]);
-  let blockheight = await connection.getBlockHeight(commitment);
-  const rawTransaction = tx.serialize();
-  let transactionSent = false;
-  while (blockheight < recentBlockInfo.lastValidBlockHeight + 15) {
-    try {
-      if (blockheight < recentBlockInfo.lastValidBlockHeight || !transactionSent) {
-        await SEND_THROTTLE(async () => {
-          await connection.sendRawTransaction(rawTransaction, {
-            maxRetries: 0,
-            minContextSlot: context.slot,
-            preflightCommitment: commitment,
-            skipPreflight: true,
-          })
-        });
-        transactionSent = true;
-      }
-    } catch (e) {
-      if (
-        transactionSent ||
-        (e instanceof SendTransactionError && e.message.includes("Minimum context slot has not been reached"))
-      ) {
-        await sleep(500);
-        continue;
-      }
-      throw e;
+    const { context, value: recentBlockInfo } = await connection.getLatestBlockhashAndContext({ commitment });
+    let tx = buildStreamTransaction(ixs, recentBlockInfo.blockhash, sender, metadata);
+    const res = await simulateTransaction(connection, tx);
+    let newCu = DEFAULT_CU;
+    if (res.value.unitsConsumed) {
+      newCu = Math.floor(res.value.unitsConsumed * (1 + COMPUTE_ERROR_MARGIN / 10_000));
     }
-    await sleep(500);
-    try {
-      const value = await confirmAndEnsureTransaction(connection, signature);
-      if (value) {
-        return { txId: signature, contractId: metadata.publicKey.toBase58() };
+    ixs[0] = ComputeBudgetProgram.setComputeUnitLimit({ units: newCu });
+    tx = buildStreamTransaction(ixs, recentBlockInfo.blockhash, sender, metadata);
```

Comment:
if `res.value.unitsConsumed` isn't defined we don't need to rebuild this tx again, do we?

there might be a reason for that if simulations take time and our used blockhash is stale. Do we need to refetch blockhash one more time?

Reply:
so in fact th flow is:
- simulate with 500k
- if `unitsConsumed` is defined - use it with the 1.1 offset
- otherwise, use default 220k